src/v/cluster/types.h

```diff
@@ -84,6 +85,8 @@ struct topic_configuration {
     storage::ntp_config make_ntp_config(
       const ss::sstring&, model::partition_id, model::revision_id) const;
 
+    bool is_internal() const { return tp_ns.ns == kafka_internal_namespace; }
```

Comment:
please add a code comment here and in the cover letter of the PR saying that cluster topics are only used at bootstrap and need to cache. 

alternatively, we can simply purge all caches after node recovery?